onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorLayerNormalization.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "precomp.h"
+
+namespace Dml
+{
+
+class DmlOperatorLayerNormalization : public DmlOperator
+{
+public:
+    DmlOperatorLayerNormalization(const MLOperatorKernelCreationContext& kernelCreationContext)
+    :   DmlOperator(kernelCreationContext)
+    {
+        std::vector<std::optional<uint32_t>> kernelInputIndices = {0, 1, 2};
+        DmlOperator::Initialize(kernelCreationContext, kernelInputIndices);
+
+        const float epsilon = kernelCreationContext.GetOptionalAttribute<float>(AttrName::Epsilon, DefaultEpsilon);
+
+        int32_t onnxAxis = kernelCreationContext.GetOptionalAttribute<int32_t>(AttrName::Axis, -1);
+        uint32_t inputDimCount = kernelCreationContext.GetTensorShapeDescription().GetInputTensorDimensionCount(0);
+        onnxAxis = OperatorHelper::HandleNegativeAxis(onnxAxis, inputDimCount);
+        std::vector<int32_t> onnxAxes(inputDimCount - onnxAxis);
+        std::iota(onnxAxes.begin(), onnxAxes.end(), onnxAxis);
+
+        std::vector<uint32_t> dmlAxes;
+        GetDmlAdjustedAxes(onnxAxes, inputDimCount, m_inputTensorDescs.front().GetDimensionCount(), /*out*/ dmlAxes);
+
```

Comment:
Let's simplify this a little and delete these 2 lines, as this is an artifact of the old 4D minimum. Notice in InstanceNorm, I passed a minimumDimension of 1 to `Initialize` which avoids the extra adjustment later:

[https://github.com/microsoft/onnxruntime/pull/12693/files](https://github.com/microsoft/onnxruntime/pull/12693/files)

Then the `onnxAxes` and `dmlAxes` are the same thing.